src/utils/MerkleProof.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity >=0.8.0;
+
+library MerkleProof {
+    function verify(
+        bytes32[] memory proof,
+        bytes32 root,
+        bytes32 leaf
+    ) internal pure returns (bool isValid) {
+         assembly {
+            let computedHash := leaf
+            let proofLength := mload(proof)
+
+            // exit early if empty proof is supplied
+            if iszero(proofLength) {
```

Comment:
probably can remove this, adds gas for the success case.

Reply:
this is true - @transmissions11 what do you think?

Reply:
As long as it still reverts in other cases I agree